jung-visualization/src/test/java/edu/uci/ics/jung/visualization/spatial/rtree/RTreeTest.java

```diff
@@ -227,11 +228,34 @@ private void testAreas(TreeNode rootNode) {
     Rectangle2D rootBounds = rootNode.getBounds();
     if (rootNode instanceof InnerNode) {
       InnerNode innerNode = (InnerNode) rootNode;
-      Assert.assertEquals(rootBounds, Node.union(innerNode.getChildren()));
+      Comparator<Rectangle2D> comparator = new CloseEnoughComparator();
+      Assert.assertTrue(comparator.compare(rootBounds, Node.union(innerNode.getChildren())) == 0);
     }
 
     for (TreeNode rt : rootNode.getChildren()) {
       testAreas(rt);
     }
   }
+
+  class CloseEnoughComparator implements Comparator<Rectangle2D> {
+
+    boolean closeEnough(double left, double right) {
+      return Math.abs(left - right) < 0.001;
+    }
+
+    @Override
+    public int compare(Rectangle2D left, Rectangle2D right) {
+      if (left.equals(right)) {
+        return 0;
+      }
+      log.info("Using the CloseEnoughComparator for {} and {}", left, right);
+      if (closeEnough(left.getMinX(), right.getMinX())
+          && closeEnough(left.getMinY(), right.getMinY())
+          && closeEnough(left.getMaxX(), right.getMaxX())
+          && closeEnough(left.getMaxY(), right.getMaxY())) {
+        return 0;
+      }
+      return -1;
+    }
+  }
```

Comment:
This hack is because the test would sometimes fail (as in your most recent push) because of comparing floating point values like this:
Failed tests:   testOne(edu.uci.ics.jung.visualization.spatial.rtree.RTreeTest): expected:<java.awt.geom.Rectangle2D$Double[x=2.9168632067831934,y=0.1405494536367291,w=**570.535341820459**,h=599.8594505463633]> but was:<java.awt.geom.Rectangle2D$Double[x=2.9168632067831934,y=0.1405494536367291,w=**570.5353418204589**,h=599.8594505463633]>

Reply:
Yep.  I think I actually expressed some concern about the direct comparison of doubles in some previous PR; I don't remember the exact context, but in general we probably don't want to be doing that.